lib/cli/src/commands/package/publish.rs

```diff
@@ -0,0 +1,152 @@
+            default_missing_value = "container",
+            value_enum
+        )]
+    pub wait: PublishWait,
+
+    /// Timeout (in seconds) for the publish query to the registry.
+    ///
+    /// Note that this is not the timeout for the entire publish process, but
+    /// for each individual query to the registry during the publish flow.
+    #[clap(long, default_value = "5m")]
+    pub timeout: humantime::Duration,
+
+    /// Whether or not the patch field of the version of the package - if any - should be bumped.
+    #[clap(long, conflicts_with = "version")]
+    pub bump: bool,
+
+    /// Do not prompt for user input.
+    #[clap(long, default_value_t = !std::io::stdin().is_terminal())]
+    pub non_interactive: bool,
+}
+
+#[async_trait::async_trait]
+impl AsyncCliCommand for PackagePublish {
+    type Output = PackageIdent;
+
+    async fn run_async(self) -> Result<Self::Output, anyhow::Error> {
+        tracing::info!("Checking if user is logged in");
+        let client = login_user(
+            &self.api,
+            &self.env,
+            !self.non_interactive,
+            "publish a package",
+        )
+        .await?;
+
+        tracing::info!("Loading manifest");
+        let (manifest_path, manifest) = get_manifest(&self.package_path)?;
+        tracing::info!("Got manifest at path {}", manifest_path.display());
+
+        let (package_namespace, package_hash) = {
+            let push_cmd = PackagePush {
+                api: self.api.clone(),
+                env: self.env.clone(),
+                dry_run: self.dry_run,
+                quiet: self.quiet,
+                package_namespace: self.package_namespace.clone(),
+                timeout: self.timeout.clone(),
+                bump: self.bump.clone(),
+                non_interactive: self.non_interactive.clone(),
+                wait: self.wait.clone(),
+                package_path: self.package_path.clone(),
+            };
+
+            push_cmd.push(&client, &manifest, &manifest_path).await?
+        };
+
+        let ident = PackageTag {
+            api: self.api.clone(),
+            env: self.env.clone(),
+            dry_run: self.dry_run.clone(),
+            quiet: self.quiet,
+            package_namespace: Some(package_namespace),
+            package_name: self.package_name.clone(),
+            package_version: self.package_version.clone(),
+            timeout: self.timeout.clone(),
+            bump: self.bump.clone(),
+            non_interactive: self.non_interactive.clone(),
+            package_path: self.package_path.clone(),
+        }
+        .tag(&client, &manifest, package_hash)
+        .await?;
+
+        tracing::info!("Proceeding to invalidate query cache..");
+
+        if let Err(e) = invalidate_graphql_query_cache(&self.env.cache_dir) {
+            tracing::warn!(
+                error = &*e,
+                "Unable to invalidate the cache used for package version queries",
+            );
+        }
```

Comment:
Why we are doing this?

Reply:
So that `wasmer run` doesn't use stale (cached) package versions after `wasmer publish`.

Reply:
Ummm... can we add that as a comment next to the if/code @xdoardo ?

Reply:
https://github.com/wasmerio/wasmer/pull/4643/commits/64e969053f7124302e86fbb97c431abd86f99fb2#diff-a33bbd632caf21f48a1c3682c20a6ab985976a62180877d94e4eea3c5d5ffc72R193